crates/fs/src/fs.rs

```diff
@@ -1999,6 +2013,70 @@ fn chunks(rope: &Rope, line_ending: LineEnding) -> impl Iterator<Item = &str> {
     })
 }
 
+fn create_temp_file(path: &Path) -> Result<NamedTempFile> {
+    let temp_file = if cfg!(any(target_os = "linux", target_os = "freebsd")) {
+        // Use the directory of the destination as temp dir to avoid
+        // invalid cross-device link error, and XDG_CACHE_DIR for fallback.
+        // See https://github.com/zed-industries/zed/pull/8437 for more details.
+        NamedTempFile::new_in(path.parent().unwrap_or(paths::temp_dir()))?
+    } else if cfg!(target_os = "windows") {
+        // If temp dir is set to a different drive than the destination,
+        // we receive error:
+        //
+        // failed to persist temporary file:
+        // The system cannot move the file to a different disk drive. (os error 17)
+        //
+        // So we use the directory of the destination as a temp dir to avoid it.
+        // https://github.com/zed-industries/zed/issues/16571
+        NamedTempFile::new_in(path.parent().unwrap_or(paths::temp_dir()))?
+    } else {
+        NamedTempFile::new()?
+    };
+
+    Ok(temp_file)
+}
+
+#[cfg(target_os = "macos")]
+async fn execute_elevated_command(_command: &str) -> Result<()> {
+    unimplemented!("execute_elevated_command is not implemented")
+}
+
+#[cfg(target_os = "windows")]
+async fn execute_elevated_command(_command: &str) -> Result<()> {
+    unimplemented!("execute_elevated_command is not implemented")
+}
+
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+async fn execute_elevated_command(command: &str) -> Result<()> {
+    let pkexec_path = Path::new("/usr/bin/pkexec");
+
+    if !pkexec_path.exists() {
+        return Err(anyhow::anyhow!(
+            "pkexec not found at {}",
+            pkexec_path.display()
+        ));
+    }
+
+    let mut cmd = Command::new(pkexec_path);
+    cmd.arg("--disable-internal-agent");
+
+    let script_path = PathBuf::from("/usr/libexec/zed/elevate.sh");
```

Comment:
I’ve added the env variable which can be used in Nix installation. But, I'm not picking that in this PR as I'm not familiar with Nix yet. It should still work on NixOS as long as pkexec is in the PATH without custom message. I will create another PR after setting up NixOS VM and testing there.

Reply:
Thanks! In nixpkgs we wrap the executable in a shell script to add node.js into its PATH already so adding pkexec in that mix is trivial. Similarly setting the environment variable for the elevate script to point to `/nix/store/<hash>` is easily doable now. Updating the nix files in this repo can indeed be done separately (especially since this is not a change that breaks something that was working before).